docs/arch.md

```diff
@@ -0,0 +1,134 @@
+    for event_record in event["Records"]:
+        message = json.loads(event_record["body"])
+        dds.delete(message['key'])
+```
+
+Each SQS event spawns a lambda function (specifically, a domovoi app) whose job is to delete the queue item(s)
+passed to it. The `dds.delete()` call also handles checking the list of files that should not be deleted (see
+below).
+
+This architecture of populating a queue with items, and removing each item(s) from the queue and triggering a
+lambda function to process the item, enables a massive scale-up of deletion events using a swarm of lambda
+functions.
+
+The Kharon delete method, called by the lambda swarm on each queue item, is in `dds/__init__.py`
+[(link)](https://github.com/HumanCellAtlas/kharon/blob/master/dds/__init__.py).
+
+```text
+def delete(key):
+    if key_in_inclusion_list(key):
+        logger.debug(f"Not deleting whitelisted {key}")
+    else:
+        for replica in BUCKETS:
+            _delete(replica, key)
+        deindex_bundle(key)
+```
+
+This is a short method that checks if the key to be deleted is in the inclusion list,
+and deletes the key if it is not.
+
+There is one deletion queue per stage.
+
+### Lambda Function Logic
+
+Each lambda function is spawned by an SQS queue event and has an item that it is tasked with deleting.
+The lambda function is responsible for checking whether the item is okay to delete, or whether it has
+been marked as a special item that should not be deleted.
+
+Here is a flowchart illustrating the lambda function logic for the `dds-delete` Domovoi app:
+
+![Architecture diagram - dds-delete lambda function logic](img/arch_lambda_logic.png)
+
+### Inclusion Lists
+
+It is important to be able to mark files to be saved, in addition to marking files to be deleted. This is the
+purpose of the inclusion list, which is stored in the cloud as a DynamoDB table. Any file, blob, or bundle in
+this list will always survive deletion.
+
+As described above, each SQS queue event spawns a lambda function to process the item(s) to be deleted.
+In the call to `dds.delete()`, the lambda function will compare the key of the file it is supposed to
+delete to the keys in the inclusion list. If the file is in the inclusion list, the lambda function will
+exit without touching the file. This architecture (using a highly scalable database to store the inclusion list)
+allows the lambda swarm to scale without creating a bottleneck with the inclusion list lookup step.
+
+To save specific items from being deleted, an inclusion list should be assembled in a text file. An operator can
+then use the script at `scripts/populate_whitelist.py`
+[(link)](https://github.com/HumanCellAtlas/kharon/blob/master/scripts/populate_whitelist.py)
+and pass the file containing the inclusion list, and each item in the inclusion list file will be added to the
+DynamoDB inclusion list.
+
+Note: it is good practice to include the stage name in the inclusion list filename.
```

Comment:
@amarjandu This is the only item that I haven't addressed, since we can discuss the example we will provide. I would prefer to create a ticket and address this in a follow-up PR, if ok with you.